Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs

```diff
@@ -127,13 +128,15 @@ string contentTypeName
 
             var componentModel = serviceProvider.GetComponentModel();
             var workspaceService = componentModel.GetService<IVsFolderWorkspaceService>();
+            var pythonWorkspaceProvider = componentModel.GetService<IPythonWorkspaceContextProvider>();
```

Comment:
So this one is always called from the ui thread?

Reply:
I'll add a call to switch to main thread to that one and the other one